File: cpp/countingValleys.c
#include <stdio.h>
#include <stdlib.h>

void countingValleys( int n, char s[] ){
    int *v, i;
    v = (int *) malloc(n * sizeof(int));
    for(i = 0; i < n; i++){

    }
}

int main(void){
    
    int n = 8;
    char r[8] = "UDDDUDUU";
    char s[8] = "DDUUUUDD";
    countingValleys(n, r);
    countingValleys(n, s);

}

File: python/countingValleys.py
def countingValleys(n, s):
    
    u = []
    d = []
    ucont = 0
    dcont = 0
    for i in range(n):
        if(s[i] == 'D'): 

            if(i < (n-1)):       
                ucont += 1
                if(s[i+1] == 'U'):
                    u.append(ucont)
                    ucont = 0
            else:
                u.append(ucont + 1)

    for j in range(n):
        if(s[j] == 'U'): 

            if(j < (n-1)):       
                dcont += 1
                if(s[j+1] == 'D'):
                    d.append(dcont)
                    dcont = 0
            else:
                d.append(dcont + 1)
    
    print(u, d)

    
        
    return 0

    
    
n = 8
r = "UDDDUDUU" #1
s = "DDUUUUDD" #2
t = "DDUUDDUDUUUD" #2

print(countingValleys(n, s))
#print("S = ", countingValleys(n, s))



#python2 countingValleys.py